pkg/solana/chainwriter/transform_registry.go

```diff
@@ -0,0 +1,110 @@
+package chainwriter
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// TODO: make this type in the chainlink-common CW package
+type ReportPreTransform struct {
+	ReportContext  [2][32]byte
+	Report         []byte
+	Info           ccipocr3.ExecuteReportInfo
+	AbstractReport ccip_router.ExecutionReportSingleChain
+}
+
+type ReportPostTransform struct {
+	ReportContext  [2][32]byte
+	Report         []byte
+	Info           ccipocr3.ExecuteReportInfo
+	AbstractReport ccip_router.ExecutionReportSingleChain
+	TokenIndexes   []byte
+}
+
+func FindTransform(id string) (func(context.Context, *SolanaChainWriterService, any, solana.AccountMetaSlice, string) (any, error), error) {
+	switch id {
+	case "CCIP":
+		return CCIPArgsTransform, nil
+	default:
+		return nil, fmt.Errorf("transform not found")
+	}
+}
+
+// This example doesn't contain the complete implementation of the function, since the
+// types needed to transform can't be imported into this repository. However, in production, this
```

Comment:
Is this outdated comment?
I didn't understand. This looks like an already complete implementation.

Reply:
Yep, that was outdated. I updated the comment